src/pages/index.tsx

```diff
@@ -156,6 +157,20 @@ export const getServerSideProps = withMultipleServerSideProps(
       throw courtCases
     }
 
+    const lastPossiblePageNumber = calculateLastPossiblePageNumber(courtCases.totalCases, +validatedMaxPageItems)
+    if (+validatedPageNum > lastPossiblePageNumber) {
+      if (req.url) {
+        const [urlPath, urlQuery] = req.url.split("?")
+        const parsedUrlQuery = new URLSearchParams(urlQuery)
+        parsedUrlQuery.set("page", lastPossiblePageNumber.toString())
+        return redirectTo(`${urlPath}?${parsedUrlQuery.toString()}`)
```

Comment:
Looks good! However, this is not covered by a test. I wonder if we could add a simple e2e test